applications/MeshingApplication/tests/cpp_tests/test_tetrahedra10_mesh_converter_utility.cpp

```diff
@@ -0,0 +1,129 @@
+        Element::Pointer tetra2 = subMp2.CreateNewElement("Element3D4N", 2, {2, 3, 4, 5}, p_properties_1);
+
+        subMp2.AddNode(modelpart.pGetNode(5));
+        subMp2.AddNode(modelpart.pGetNode(2));
+        subMp2.AddNode(modelpart.pGetNode(3));
+        subMp2.AddNode(modelpart.pGetNode(4));
+        subMp1.AddNode(modelpart.pGetNode(1));
+
+    
+        std::vector<double> volumes(4); 
+        GeometryPtrType geom1 = tetra1->pGetGeometry();
+        volumes[tetra1->Id()] = geom1->Volume();
+        GeometryPtrType geom2 = tetra2->pGetGeometry();
+        volumes[tetra2->Id()] = geom2->Volume(); 
+
+        Condition::Pointer cond1 = subMp2.CreateNewCondition("SurfaceCondition3D3N", 3, {2, 3, 5}, p_properties_1);
+
+        GeometryPtrType geom3 = cond1->pGetGeometry();
+        volumes[cond1->Id()] = geom3->Area(); 
+
+        Tetrahedra10MeshConverter refineTetra(modelpart); 
+        refineTetra.LocalConvertTetrahedra10Mesh(false,false);
+
+        KRATOS_CHECK_EQUAL(modelpart.Nodes().size(),14); //There are 14 nodes (10 for each tetra but 6 are shared)
+        KRATOS_CHECK_EQUAL(subMp0.Nodes().size(),14); //Also in the first level submodelpart
+        KRATOS_CHECK_EQUAL(subMp1.Nodes().size(),14); //Also in the second level submodelpart
+        KRATOS_CHECK_EQUAL(subMp2.Nodes().size(),10); //In the third level submodelpart only 10 nodes
+        KRATOS_CHECK_EQUAL(modelpart.Elements().size(),2); //No new elements are added
+        KRATOS_CHECK_EQUAL(subMp1.Elements().size(),2); //No new elements are added
+        KRATOS_CHECK_EQUAL(subMp2.Elements().size(),1); //No new elements are added
+        KRATOS_CHECK_EQUAL(subMp2.Conditions().size(),1); //No new conditions are added
+        
+        for(auto elem : modelpart.Elements()) {
+            GeometryPtrType geom = elem.pGetGeometry();
+            const auto geometryType = geom->GetGeometryType();
+            KRATOS_CHECK_EQUAL(geometryType, GeometryData::KratosGeometryType::Kratos_Tetrahedra3D10);
+            const double vol = geom->Volume();
+            KRATOS_CHECK_NEAR(volumes[elem.Id()], vol ,0.001);
+
+            auto points = geom->Points();
+            KRATOS_CHECK_EQUAL(Distance(points[0],points[1]), Distance(points[0],points[4]) + Distance(points[4],points[1]) );
+            KRATOS_CHECK_EQUAL(Distance(points[1],points[2]), Distance(points[1],points[5]) + Distance(points[5],points[2]) );
+            KRATOS_CHECK_EQUAL(Distance(points[2],points[3]), Distance(points[2],points[9]) + Distance(points[9],points[3]) );
+            KRATOS_CHECK_EQUAL(Distance(points[3],points[0]), Distance(points[3],points[7]) + Distance(points[7],points[0]) );
+            KRATOS_CHECK_EQUAL(Distance(points[0],points[2]), Distance(points[0],points[6]) + Distance(points[6],points[0]) );
+            KRATOS_CHECK_EQUAL(Distance(points[1],points[3]), Distance(points[1],points[8]) + Distance(points[8],points[1]) );
```

Comment:
Considering that `Distance` function does floating point operations, I'd use `KRATOS_CHECK_NEAR` with a very strict tolerance instead.

Reply:
Totally right